etp-backend/src/main/clj/solita/etp/service/palveluvayla.clj

```diff
@@ -0,0 +1,41 @@
+(ns solita.etp.service.palveluvayla
+  (:require [solita.etp.service.energiatodistus :as service.energiatodistus]
+            [solita.etp.service.energiatodistus-pdf :as service.energiatodistus-pdf]
+            [solita.etp.service.energiatodistus-search :as service.energiatodistus-search]))
+
+(def i-am-paakayttaja {:rooli 2})
+
+(defn find-first-existing-pdf
+  "Return the first language version that exists if any. If language-preference-order is not given, [fi sv] is used."
+  [id language-preference-order db aws-s3-client]
+  (some identity (->> (or language-preference-order ["fi" "sv"])
+                      (map #(service.energiatodistus-pdf/find-energiatodistus-pdf db
```

Comment:
Tässä voisi ehkä tuon mapin korvata jollakin ei-laiskalla funktiolla, jotta energiatodistuksen haku blob storagesta tapahtuu varmasti ennakoitavaan aikaan koodissa.

(some identity) voisi myös olla threading-makrossa?

Reply:
Ideana oli hakea niistä vain eka joka löytyy. Clojurehan ei niin tee, mutta ajattelin että kommunikoi paremmin haetun toiminnallisuuden jos niin tekee. 

(some identity) toimii myö makrossa, mutta taas minusta näin päin tuntuu kommunikoivan tarkoituksen paremmin.